client/app/telecom/telephony/alias/dashboard/telecom-telephony-alias-dashboard.controller.js

```diff
@@ -0,0 +1,180 @@
+    function extractIncomingCallsDetails(calls) {
+      return calls.filter(call => call.wayType !== 'outgoing')
+        .map((call) => {
+          _.set(call, 'durationAsDate', new Date(call.duration * 1000));
+          return call;
+        });
+    }
+
+    function extractOutgoingCallsDetails(calls) {
+      return calls.filter(call => call.wayType !== 'incoming' && call.duration > 0)
+        .map((call) => {
+          _.set(call, 'durationAsDate', new Date(call.duration * 1000));
+          return call;
+        });
+    }
+
+    return this.voipService.getServiceConsumption(this.number)
+      .then((conso) => {
+        const incomingCalls = extractIncomingCallsDetails(conso);
+        const outgoingCalls = extractOutgoingCallsDetails(conso);
+
+        this.consumption.incoming = {
+          total: incomingCalls.length,
+          duration: new Date(_.sum(incomingCalls, call => call.duration) * 1000),
+        };
+
+        this.consumption.outgoing = {
+          total: outgoingCalls.length,
+          duration: new Date(_.sum(outgoingCalls, call => call.duration) * 1000),
+          outplan: _.round(_.sum(
+            outgoingCalls.filter(call => call.planType === 'outplan' && call.priceWithoutTax),
+            call => call.priceWithoutTax.value,
+          ), 2),
+        };
+
+        this.buildConsumptionChart(incomingCalls, outgoingCalls);
+      });
+  }
+
+  buildConsumptionChart(incomingCalls, outgoingCalls) {
+    const datasetConfiguration = {
+      dataset: {
+        fill: true,
+        borderWidth: 1,
+      },
+    };
+
+    const _incomingCalls = incomingCalls.map(call => (
+      { callDate: moment(call.creationDatetime).format('YYYY-MM-DD').toString() }
+    ));
+
+    const _outgoingCalls = outgoingCalls.map(call => (
+      { callDate: moment(call.creationDatetime).format('YYYY-MM-DD').toString() }
+    ));
+
+    const xAxisKeys = _.chain(_incomingCalls)
+      .union(_outgoingCalls)
+      .sortBy('callDate')
+      .groupBy('callDate')
+      .keys()
+      .value();
+
+    function convertCallsToChartData(calls) {
+      const groupedCalls = _.groupBy(calls, 'callDate');
+
+      return xAxisKeys.map(key => ({
+        x: key,
+        y: groupedCalls[key] ? groupedCalls[key].length : 0,
+      }));
+    }
+
+    this.consumption.chart.addSerie(
+      this.$translate.instant('telephony_alias_consumption_outgoing_calls'),
+      convertCallsToChartData(_outgoingCalls), datasetConfiguration,
+    );
+
+    this.consumption.chart.addSerie(
+      this.$translate.instant('telephony_alias_consumption_incoming_calls'),
+      convertCallsToChartData(_incomingCalls), datasetConfiguration,
+    );
+  }
+
+  deleteConfiguration() {
+    this.$uibModal.open({
+      templateUrl: '/app/telecom/telephony/alias/dashboard/confirmDeleteConfiguration/confirmDeleteConfiguration.modal.html',
+      controller: 'TelecomTelephonyAliasConfirmDeleteConfigurationCtrl',
+      controllerAs: '$ctrl',
```

Comment:
Isn't that the default name of the controller ?

Reply:
no, it's not the same as `component` apparently